auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/AdminEnableDisableUserController.kt

```diff
@@ -0,0 +1,101 @@
+package uk.gov.communities.delta.auth.controllers.internal
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.util.*
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.AzureADSSOConfig
+import uk.gov.communities.delta.auth.config.DeltaConfig
+import uk.gov.communities.delta.auth.config.LDAPConfig.Companion.VALID_USERNAME_REGEX
+import uk.gov.communities.delta.auth.plugins.ApiError
+import uk.gov.communities.delta.auth.repositories.LdapUser
+import uk.gov.communities.delta.auth.services.SetPasswordTokenService
+import uk.gov.communities.delta.auth.services.UserAuditService
+import uk.gov.communities.delta.auth.services.UserLookupService
+import uk.gov.communities.delta.auth.services.UserService
+import uk.gov.communities.delta.auth.utils.randomBase64
+import javax.naming.NameNotFoundException
+
+class AdminEnableDisableUserController(
+    private val ssoConfig: AzureADSSOConfig,
+    private val userLookupService: UserLookupService,
+    private val userService: UserService,
+    private val setPasswordTokenService: SetPasswordTokenService,
+    private val auditService: UserAuditService,
+) : AdminUserController(userLookupService) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    private val allowedRoles =
+        arrayOf(DeltaConfig.DATAMART_DELTA_READ_ONLY_ADMIN, DeltaConfig.DATAMART_DELTA_ADMIN)
```

Comment:
"Read only" admin shouldn't be allowed to this should they?

Reply:
Good question, in practice "Read only admin" is assigned to the helpdesk (Excela). They definitely can and do send activation emails, which lets users enable their own accounts, so I think they may as well be able to directly enable and disable accounts. We definitely need to change the name of the role though.

Reply:
Cool. Maybe a comment against the enum for now or something.